jmxtrans-output/jmxtrans-output-influxdb/src/main/java/com/googlecode/jmxtrans/model/output/InfluxDbWriter.java

```diff
@@ -0,0 +1,307 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.googlecode.jmxtrans.model.output;
+
+import java.lang.reflect.Method;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.lang.StringUtils;
+import org.influxdb.InfluxDB;
+import org.influxdb.InfluxDB.ConsistencyLevel;
+import org.influxdb.InfluxDBFactory;
+import org.influxdb.dto.BatchPoints;
+import org.influxdb.dto.Point;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import com.googlecode.jmxtrans.model.Query;
+import com.googlecode.jmxtrans.model.Result;
+import com.googlecode.jmxtrans.model.Server;
+import com.googlecode.jmxtrans.model.ValidationException;
+
+/**
+ * {@link com.googlecode.jmxtrans.model.OutputWriter} for
+ * <a href="https://influxdb.com/index.html">InfluxDB</a>.
+ *
+ * @author Simon Hutchinson <https://github.com/sihutch>
+ */
+public class InfluxDbWriter extends BaseOutputWriter {
+
+	/**
+	 * Enumerates the members of {@link Result} that may be written as
+	 * {@link Point} tags
+	 * 
+	 * @author Simon Hutchinson <https://github.com/sihutch>
+	 *
+	 */
+	public enum ResultAttribute {
+
+		TYPENAME("typeName"), OBJDOMAIN("objDomain"), CLASSNAME("className"), ATTRIBUTENAME("attributeName");
+
+		private String tagName;
+		private String accessorMethod;
+
+		ResultAttribute(String tagName) {
+			this.tagName = tagName;
+			this.accessorMethod = "get" + StringUtils.capitalize(tagName);
+		}
+
+		public String getTagName() {
+			return tagName;
+		}
+
+		public String getAccessorMethod() {
+			return accessorMethod;
+		}
+	}
+
+	/**
+	 * Setting name that allows attributes from {@link Result} limited by
+	 * {@link ResultAttribute} values to be written as {@link Point} tags
+	 */
+	public static final String SETTING_RESULT_TAGS = "resultTags";
+
+	/**
+	 * The {@link EnumSet} of {@link ResultAttribute} attributes of
+	 * {@link Result} that will be written as {@link Point} tags
+	 */
+	private EnumSet<ResultAttribute> resultAttributesToWriteAsTags = EnumSet.allOf(ResultAttribute.class);
```

Comment:
Note that Guava provides an immutable EnumSet : http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/collect/Sets.html#immutableEnumSet(java.lang.Iterable)